adds support for Spring Boot 3 in appsso-starter-java accelerator

ddubson opened the pull request:
Applies to: TAP 1.5.x only

trisberg commented on appsso-starter-java/build.gradle:
```diff
@@ -19,13 +21,18 @@ dependencies {
 
+tasks.named("bootBuildImage") {
+	environment["BP_JVM_VERSION"] = "17"
+	buildpacks = ["urn:cnb:builder:paketo-buildpacks/java"]
```
I'm curious why you need to specify this

ddubson replied:
Removed.

DiegoWong reviewed (approved):
I tested creating a project from this accelerator and running ./gradlew build, both things worked fine

trisberg commented:
Yes, I think we need at least a `contextLoads` type test for this sample

trisberg reviewed (changes requested):
I think we need the test added and the missing bean configuration fixed

ddubson commented:
@trisberg I will remove bootBuildImage task as it is not **inherently** needed, but nice to have.

@DiegoWong - you may run into issues running this project by itself as it needs ClientRegistration details, that on TAP are projected to the Workload Pod's. However, I will add an integration test to simulate this.

ddubson commented:
@trisberg @DiegoWong 

I've update the commit with the following:

1. Added `AppSSOAcceleratorApplicationTest` which exercises loading the application and navigating to the home page.
2. Added an `application-local.yml` for development purposes (or as an example of how to configure the application for running in development mode). You _still_ need to have an OAuth authorization server to run the accelerator in `local` mode however, but that is the nature of this accelerator, it is inherently dependent on external authn/authz.

trisberg commented:
Trying to deploy this but the docs have me confused. Should this sample point to the latest 1.5 docs instead of 1.4?

This command from the 1.4 docs failed
```
% ytt \
  --file client.yaml \
  --data-value namespace=workloads \
  --data-value workload_name=appsso-starter-java \
  --data-value domain=127.0.0.1.nip.io \
  --data-value auth_server_name="" \
  --data-value claim_name=appsso-starter-java | \
   kubectl apply -f-

ytt: Error: 
- struct has no .authserver_label field or method (did you mean .auth_server_name?)
    in <toplevel>
      client.yaml:10 |     matchLabels: #@ data.values.authserver_label
error: no objects passed to apply
```

while this from 1.5 draft seems to work
```
% ytt \          
  --file client.yaml \
  --data-value namespace=workloads \
  --data-value workload_name=appsso-starter-java \
  --data-value domain=127.0.0.1.nip.io \
  --data-value authserver_label.my-sso=true \
  --data-value authserver_label.env=dev |
   kubectl apply -f-

clientregistration.sso.apps.tanzu.vmware.com/appsso-starter-java created
```

trisberg commented:
I'm getting an error when deploying the generated app

```
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.926925250-05:00 Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949861495-05:00 2023-02-22 16:26:40.949 ERROR 110 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949897195-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949903595-05:00 ***************************
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949907495-05:00 APPLICATION FAILED TO START
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949910995-05:00 ***************************
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949914495-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949918196-05:00 Description:
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949921496-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949925896-05:00 Field clientRegistrationRepository in com.vmware.tanzu.apps.sso.accelerator.config.WebSecurityConfig required a bean of type 'org.springframework.security.oauth2.client.registration.ClientRegistrationRepository' that could not be found.
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949940796-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949945396-05:00 The injection point has the following annotations:
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949949496-05:00 	- @org.springframework.beans.factory.annotation.Autowired(required=true)
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949952896-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949956296-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949959896-05:00 Action:
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949963296-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949967496-05:00 Consider defining a bean of type 'org.springframework.security.oauth2.client.registration.ClientRegistrationRepository' in your configuration.
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949970896-05:00 
appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:41.023586902-05:00 [[Command exited with 1]]
```

ddubson commented:
> I'm getting an error when deploying the generated app
> 
> ```
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.926925250-05:00 Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949861495-05:00 2023-02-22 16:26:40.949 ERROR 110 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949897195-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949903595-05:00 ***************************
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949907495-05:00 APPLICATION FAILED TO START
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949910995-05:00 ***************************
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949914495-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949918196-05:00 Description:
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949921496-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949925896-05:00 Field clientRegistrationRepository in com.vmware.tanzu.apps.sso.accelerator.config.WebSecurityConfig required a bean of type 'org.springframework.security.oauth2.client.registration.ClientRegistrationRepository' that could not be found.
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949940796-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949945396-05:00 The injection point has the following annotations:
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949949496-05:00 	- @org.springframework.beans.factory.annotation.Autowired(required=true)
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949952896-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949956296-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949959896-05:00 Action:
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949963296-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949967496-05:00 Consider defining a bean of type 'org.springframework.security.oauth2.client.registration.ClientRegistrationRepository' in your configuration.
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:40.949970896-05:00 
> appsso-starter-java-00001-deployment-845f4cdb9b-hxnk4[workload] 2023-02-22T11:26:41.023586902-05:00 [[Command exited with 1]]
> ```

Do you have a `ClientRegistration` and `ResourceClaim` created for the `Workload`?

trisberg commented:
I have these

ddubson commented:
Looks like your ClientRegistration resource cannot match to a valid `AuthServer`, make sure your `matchLabels` in your ClientRegistration are valid labels on your AuthServer.

trisberg commented:
I think the dos are confusing since they use "default" namespace at the start but then seems to switch to a "workloads" namespace for later steps

ddubson commented:
Gotcha. We will update the docs to make them more consistent
